File: product-consumer-service/src/main/java/de/hdi/product_consumer_service/model/OrderItem.java
package de.hdi.product_consumer_service.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OrderItem {
    private Integer id;
    private Integer productId;
    private Integer quantity;
    private Double price;
    private Product product;
}

File: ts-order-service/src/models/order-item.ts
export class OrderItem {
    productId: number;
    productName: string;
    price: number;
    quantity: number;

    constructor(productId: number, productName: string, price: number, quantity: number) {
        this.productId = productId;
        this.productName = productName;
        this.price = price;
        this.quantity = quantity;
    }
}